src/libraries/System.Linq/src/System/Linq/OrderedEnumerable.SpeedOpt.cs

```diff
@@ -247,4 +247,21 @@ private TElement Last(Buffer<TElement> buffer)
             return value;
         }
     }
+
+    internal sealed partial class OrderedImplicitlyStableEnumerable<TElement> : OrderedEnumerable<TElement>
+    {
+        public override TElement[] ToArray()
+        {
+            TElement[] array = _source.ToArray();
+            Sort(array, _comparer, _descending);
+            return array;
+        }
+
+        public override List<TElement> ToList()
+        {
+            List<TElement> list = _source.ToList();
+            Sort(CollectionsMarshal.AsSpan(list), _comparer, _descending);
```

Comment:
It appears you're intentionally avoiding a `_source.ToArray()` call here. Any particular reason why?

Reply:
This method is responsible for returning a list.  I'm not sure why we'd go through an array to do it?

Reply:
Ah, nevermind then.